sql-plugin/src/main/scala/org/apache/spark/sql/rapids/AggregateFunctions.scala

```diff
@@ -658,47 +658,72 @@ case class GpuLast(child: Expression, ignoreNulls: Boolean)
+    with GpuAggregateWindowFunction[T] {
 
-  def this(child: Expression) = this(child, 0, 0)
+  def childExpression: Expression
 
-  // Both `CollectList` and `CollectSet` are non-deterministic since their results depend on the
+  // Collect operations are non-deterministic since their results depend on the
   // actual order of input rows.
   override lazy val deterministic: Boolean = false
 
   override def nullable: Boolean = false
 
-  override def prettyName: String = "collect_list"
+  override def dataType: DataType = ArrayType(childExpression.dataType, false)
 
-  override def dataType: DataType = ArrayType(child.dataType, false)
-
-  override def children: Seq[Expression] = child :: Nil
+  override def children: Seq[Expression] = childExpression :: Nil
 
   // WINDOW FUNCTION
-  override val windowInputProjection: Seq[Expression] = Seq(child)
-
-  override def windowAggregation(
-      inputs: Seq[(ColumnVector, Int)]): AggregationOnColumn[CollectListAggregation] =
-    Aggregation.collectList().onColumn(inputs.head._2)
+  override val windowInputProjection: Seq[Expression] = Seq(childExpression)
 
-  // Declarative aggregate. But for now 'CollectList' does not support it.
-  // The members as below should NOT be used yet, ensured by the
-  // "TypeCheck.aggNotGroupByOrReduction" when trying to override the expression.
-  private lazy val cudfList = AttributeReference("collect_list", dataType)()
-  // Make them lazy to avoid being initialized when creating a GpuCollectList.
+  // Make them lazy to avoid being initialized when creating a GpuCollectOp.
   override lazy val initialValues: Seq[GpuExpression] = throw new UnsupportedOperationException
   override lazy val updateExpressions: Seq[Expression] = throw new UnsupportedOperationException
   override lazy val mergeExpressions: Seq[GpuExpression] = throw new UnsupportedOperationException
   override lazy val evaluateExpression: Expression = throw new UnsupportedOperationException
-  override val inputProjection: Seq[Expression] = Seq(child)
-  override def aggBufferAttributes: Seq[AttributeReference] = cudfList :: Nil
+  override val inputProjection: Seq[Expression] = Seq(childExpression)
+
+  override def aggBufferAttributes: Seq[AttributeReference] = {
+    throw new UnsupportedOperationException
+  }
+}
+
+/**
+ * Collects and returns a list of non-unique elements.
+ *
+ * The two 'offset' parameters are not used by GPU version, but are here for the compatibility
+ * with the CPU version and automated checks.
+ */
+case class GpuCollectList(
+    childExpression: Expression,
+    mutableAggBufferOffset: Int = 0,
+    inputAggBufferOffset: Int = 0)
+    extends GpuCollectBase[CollectListAggregation] {
+
+  override def prettyName: String = "collect_list"
+
+  override def windowAggregation(
+      inputs: Seq[(ColumnVector, Int)]): AggregationOnColumn[CollectListAggregation] = {
+    Aggregation.collectList().onColumn(inputs.head._2)
```

Comment:
Setting the parameters explicitly would be better than relying on the default behavior of `collectList()`. Since it may be broken if any change in `collectList()` unexpectedly.
```suggestion
    Aggregation.collectList(NullPolicy.EXCLUDE).onColumn(inputs.head._2)
```
Yes I used the default one before but now I think it is not a good idea.